server/utils/passport/ensure-authenticated.js

```diff
@@ -9,25 +9,23 @@ export default function ensureAuthenticated(req, res, next) {
     return res.redirect(`${basePath}/logout`)
   }
   const { appDb } = req.app.locals
-  appDb
-    .collection('users')
-    .findOne(
-      { uid: req.user },
-      { 
-        _id: 0, 
-        access: 1, 
-        blocked: 1, 
-        role: 1 
+  appDb.users
+    .findUniqueOrThrow({
+      where: { uid: req.user },
+      select: {
+        access: true,
+        blocked: true,
+        role: true,
       },
-      function (err, data) {
-        switch(true) {
-          case !data || Object.keys(data).length === 0 || !!data.blocked || err:
-            return res.redirect(logoutForbiddenRoute)
-          case data.access && data.access.length === 0:
-            return res.redirect(unauthorizedRoute)
-          default:
-            return next()
-        }
+    })
+    .then((data) => {
```

Comment:
should we use async await here? Looks like we're converting the other promises to use this syntax.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/232/commits/9c2ee658fa1d23719239cab98d359a8b7e8bf7d5